### ISSUE TITLE
AttributeError: 'module' object has no attribute 'urlretrieve'

### ISSUE THREAD

**Issue by benhosmer** [roadsideseb/virtual-node#9]
From within an activated virtualenv:

`(myvenv) $ pip install virtual-node`

Python Version:
Python 3.4.3

It appears this module isn't compatible with Python 3?

**Comment by roadsideseb** [comment]
That's correct. Initially, it seemed that making the `urlretrieve` import work in a compatible way would fix the issue. Unfortunately it's not that simple. The NodeJS source package uses a large number of Python scripts to configure and build NodeJS. **But** they all only work on Python 2.x and are **not Python 3.x** compatible.
I did some brief experiments with "fixing them" using `2to3` but that would have been too simple. Several incompatibilities can't be fixed through `2to3`. 

I'm not sure how much value there is in trying to make this module Python 3 compatible if the underlying tools that are required, don't support it. However, if somebody wants to spend the time to find a solution, I'm happy to review and merge it.

Sorry for the bad news 😭.

**Comment by benhosmer** [comment]
What about [six](https://pypi.python.org/pypi/six)?

I have a use case for this module for Python 3:

Jupyterhub requires Python 3, but also requires a node module. It would be great to have a pure python setup and configuration.
